pkg/dependency/parser/dart/pub/parse.go

```diff
@@ -64,3 +91,30 @@ func (p Parser) relationship(dep string) ftypes.Relationship {
 	}
 	return ftypes.RelationshipUnknown
 }
+
+// firstVersionOfConstrain returns the first acceptable version for constraint
+func firstVersionOfConstrain(constraint string) string {
+	// cf. https://dart.dev/tools/pub/dependencies#traditional-syntax
+	switch {
+	case strings.HasPrefix(constraint, ">="):
+		constraint = strings.TrimPrefix(constraint, ">=")
+		constraint, _, _ = strings.Cut(constraint, " ")
+		return constraint
+	case strings.HasPrefix(constraint, "^"):
+		return strings.TrimPrefix(constraint, "^")
+	}
+	return ""
+}
```

Comment:
I think we want to parse the constraints in a more stable way. What if we export `constraint` and add methods like `func (c *Constraint) Version() string` and `func (c *Constraint) Operator() string`?
https://github.com/aquasecurity/go-version/blob/637058cfe4921e395b56f195224538d7eac61520/pkg/version/constraint.go#L55-L59

```
type Constraint struct {
	version  Version
	operator string // e.g. "=", ">=", "^"
	operatorFunc operatorFunc
	original string
}

func (c *Constraint) Version() string {
    return c.version
}

func (c *Constraint) Operator() string {
    return c.operator
}
```

Reply:
Merged
https://github.com/aquasecurity/go-version/pull/6

Reply:
Updated this PR -https://github.com/aquasecurity/trivy/pull/6239/commits/91c2b1b63aa0b7f607ef97a0db5ed3e42c43c32e